[WIP]Add logs to alert-details page

sangupta1 opened the pull request:
This PR restructures the `Alert details` page to have logs for each `Alert`.
Upon clicking a particular `Alert`, that `Alert` opens up. It would contain a horizontal Navbar with 2 navbar options- `Details` and `Logs`.
The `Details` option will show the details for the respective `alert`. 
The `Logs` option will show logs for the specific `Pod` and `Namespace` which is affected by the selected `Alert`. These logs are fetched from the `Log Exploration Api` filtered upon a specific time duration and number of log lines referred to as `Max Logs`. The time duration and the `Max Logs` are provided as inputs which the user can provide to get logs filtered as per these parameters. If an Alert does not have a `Pod` and `Namespace` associated with it, then the `Logs` option will show a message `No alert logs found`. If Logging is not installed or the `Log Exploration Api` is not accessible, then only the `Details` navbar option is shown and the `Logs` navbar option is not shown.
JIRA issue- https://issues.redhat.com/browse/LOG-1192

The screenshots for the changes this PR is making are- 
1.) The `Alert details` Page having 2 navbar options- `Details` and `Logs`. 
![Screenshot from 2021-05-17 19-51-18](https://user-images.githubusercontent.com/77098226/118591686-38d5b480-b7c2-11eb-883f-c7aad258b864.png)

2.) The Logs displayed by going to the `Logs` option. By default, 1000 logs from the specific `Pod` affected by the Alert are shown. These logs are returned by the `Log Exploration Api`. (In the screenshot is the `Update Available` alert which has a `Pod` and `namespace` affected by this alert. 
![Screenshot from 2021-06-03 16-34-58](https://user-images.githubusercontent.com/77098226/120635102-b3bcf180-c489-11eb-87c8-6d55064a0016.png)

3.) In this screenshot the `Update Available` alert is shown. Here the time duration selected is 0 hours and 30 minutes. It means the logs will be fetched for the duration of 30 minutes before the `Alert` is activated. The `Max Logs` entered are 7 which means 7 log lines will be fetched from the `Log Exploration Api`.
![Screenshot from 2021-06-03 17-10-50](https://user-images.githubusercontent.com/77098226/120639147-ace4ad80-c48e-11eb-84a3-182047bf30b7.png)


4.) In this screenshot the `AlertmanagerReceiversNotConfigured` Alert is shown. It does not have a `Pod` and `Namespace` linked to it. (`Watchdog` is another such alert). Hence on moving to the `Logs` tab for this alert , a message `No alert logs found` is shown.
![Screenshot from 2021-05-17 19-51-46](https://user-images.githubusercontent.com/77098226/118592155-1f813800-b7c3-11eb-9e2b-dceaaa04a37d.png)
![Screenshot from 2021-05-20 16-27-38](https://user-images.githubusercontent.com/77098226/118967481-640a0080-b988-11eb-8660-1c3013e2a228.png)



The gif for the Changes for the Administrator perspective-
![Changes_for_administrator_perspective](https://user-images.githubusercontent.com/77098226/120642587-b7a14180-c492-11eb-89f3-f6be2967c7cd.gif)


The gif for the Changes for the Developer perspective-
![Changes_for_Developer_perspective](https://user-images.githubusercontent.com/77098226/120642647-c982e480-c492-11eb-85c8-e9d6c50bc3b4.gif)

The steps for configuring the Log Exploration API are in the attached doc file-
https://docs.google.com/document/d/110I_pEMDL6DljdrZmSFMsneHn4ZmnBKXT_O2juHJrjg/edit?usp=sharing

divyanshiGupta commented:
@sangupta1 re-title your PR with a more appropriate title.

sangupta1 commented:
`/retest`

sangupta1 commented:
/retest

divyanshiGupta commented:
@sangupta1 it seems like there is a build error. Run `build.sh` and see what the error is and try to fix it. Also make sure you have the latest dependencies and your code is compiled without any errors.

divyanshiGupta commented:
/re-title [WIP]Add logs to alert-details page

divyanshiGupta commented:
/retitle [WIP]Add logs to alert-details page

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs-utils.ts:
```diff
@@ -0,0 +1,71 @@
+    loadError,
+    rule,
+  };
+};
```
Here you are importing from alert-logs file and in alert-logs file you are importing from here so it might create cyclic dependency. Move this function to alert-logs file.

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.tsx:
```diff
@@ -0,0 +1,105 @@
+  containersToStatuses,
+} from './alert-logs-utils';
+
+const AlertLogs = (props: AlertLogsProps) => {
```
```suggestion
const AlertLogs: React.FC<AlertLogsProps> = (props) => {
```

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.tsx:
```diff
@@ -0,0 +1,105 @@
+          resourceStatus={currentContainerStatus}
+        />
+      ) : (
+        <div>No logs for this Alert</div>
```
```suggestion
       <EmptyBox label="alert logs" />
```

divyanshiGupta commented on frontend/public/components/monitoring/alert-logs.tsx:
```diff
@@ -0,0 +1,105 @@
+
+export type AlertLogsProps = {
+  alert: Alert;
+  obj: ResourceKindAlert;
```
Also move these type definitions above the component.

divyanshiGupta commented:
@sangupta1 looks like you need to run `yarn i18n`.

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -324,115 +332,104 @@ export const StateCounts: React.FC<{ alerts: PrometheusAlert[] }> = ({ alerts })
+      all the listed values or regular expressions.
+    </dd>
+  </dl>
+);
```
And why have you converted these components into functions? I dont think so there is a need to do that.

divyanshiGupta replied:
If you did this just to export them then you export the component itself.

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -324,115 +332,104 @@ export const StateCounts: React.FC<{ alerts: PrometheusAlert[] }> = ({ alerts })
+    <dd>The alert has no defined severity.</dd>
+    <dd>You can also create custom severity definitions for user workload alerts.</dd>
+  </dl>
+);
```
Same as above

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -324,115 +332,104 @@ export const StateCounts: React.FC<{ alerts: PrometheusAlert[] }> = ({ alerts })
+      observability into your own services.
+    </dd>
+  </dl>
+);
```
Same as above

sangupta1 replied:
@divyanshiGupta I did not convert them into functions. I am sorry. I think during rebase with console/master and resolving merge conflict, i did this mistake and accepted an older version. I will rebase and fix this.

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -670,178 +667,92 @@ const getSourceKey = (source) => {
+          {
+            href: 'logs',
+            queryParams,
+            // t('details-page~Logs')
```
```suggestion
            // t('public~Logs')
```

divyanshiGupta commented on frontend/public/components/monitoring/alerting.tsx:
```diff
@@ -670,178 +667,92 @@ const getSourceKey = (source) => {
+          {
+            href: '',
+            queryParams,
+            // t('details-page~Details')
```
```suggestion
            // t('public~Details')
```

divyanshiGupta commented on frontend/public/locales/en/details-page.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "Details": "Details",
+  "Logs": "Logs"
+}
```
It seems it's not used anywhere now. Run yarn i18n and that should remove these strings. Also remove this file.

divyanshiGupta commented:
@sangupta1 update your screenshots with the current changes. And if possible add a gif as well.

sangupta1 commented:
@divyanshiGupta I added the relevant gifs for the changes.

divyanshiGupta commented:
Note: The feature being added by this PR is targeted for 4.9.

divyanshiGupta commented:
/kind feature

Note: The feature being added by this PR is targeted for 4.9.

divyanshiGupta commented:
@Sanyam Gupta we can keep your PR in WIP and add the Log exploration api integration changes in this PR only. Also make sure to add all the setup steps required for accessing the API in PR description and the necessary gifs/screenshots.

sangupta1 commented:
@divyanshiGupta I have updated the PR with the latest changes and added all the steps for the configuration of the API as well. Also updated all the GIF's. One of the tests is failing, i am working towards fixing it.

divyanshiGupta commented:
@sangupta1 also add the link for the backend service this PR is blocked by and also mention that after merging the backend PR this PR can be merged.
